src/Payum/Paypal/ExpressCheckout/Nvp/Action/CancelAction.php

```diff
@@ -41,9 +37,12 @@ public function execute($request)
      */
     public function supports($request)
     {
-        return
-            $request instanceof Cancel &&
-            $request->getModel() instanceof \ArrayAccess
-        ;
+        if (false == ($request instanceof Cancel && $request->getModel() instanceof \ArrayAccess)) {
+            return false;
+        }
+
+        // it must take into account only common payments, recurring payments must be cancelled by another action.
+        $model = $request->getModel();
+        return isset($model['TRANSACTIONID']);
```

Comment:
This is a BC break. before if you execute cancel request with the model that does not have a TRANSACTIONID you'll simply do nothing. Now it leads to not supported action exception. 

I'd suggest putting TRANSACTIONID back where it was and add a check `BILLINGPERIOD` is empty to supports method. 

btw could you please move this class out of Api folder. to a parent one while you are on it https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ExpressCheckout/Nvp/Action/Api/CancelRecurringPaymentsProfileAction.php